### ISSUE TITLE
Wrong line feed is used - possibly producing invalid robot syntax

### ISSUE THREAD

**Issue by yahman72** [MarketSquare/robotframework-tidy#462]
Robotidy does not seem to obey the used line feed.
I have robot file using 'LF' and that should remain like that, regardless of the OS that is used in development.

When I have a file with 'LF' 
![image](https://user-images.githubusercontent.com/8724383/219368579-e0488247-846e-46a4-9ad3-2552f8b2e18c.png)

... but when doing development in windows robotidy changes the line feed to 'CRLF'
![image](https://user-images.githubusercontent.com/8724383/219369014-cd13e925-7b41-4fa8-a998-932f04fb4b4c.png)

the fix: robotidy should use the existing line feeds and not convert them.

**Comment by yahman72** [comment]
Not sure if related  - I also have now unsupported syntax in the files that robotidy formatted.
```
Return From Keyword If      ${SKIP_THIS}      ${False}
```
becomes 
```
IF    "${s}"=="FAIL"    RETURN    ${False}
```
which triggers "IF has no closing END." robot error

So far I have only seen this in files where the linefeed conversion was also happening.

**Comment by bhirsz** [comment]
Yeah, the problem with working with line feeds is that it's hard to maintain them (especially if we're transforming whole files and they tend to get missing). That's why we convert all line feeds into common one - by default it's the native for the platform you're on. This will stay but we have added extra option that allows you to maintain existing line feeds - by using ``--lineseparator`` option:

```
--lineseparator        -ls  [native|windows|unix|auto]  Line separator to use in the outputs:                       │
│                                                                                                                     │
│                                                          • native:  use operating system's native line endings      │
│                                                          • windows: use Windows line endings (CRLF)                 │
│                                                          • unix:    use Unix line endings (LF)                      │
│                                                          • auto:    maintain existing line endings (uses what's     │
│                                                            used in the first line)                                  │
│                                                         [default: native]
```

In short, by adding ``--lineseparator auto`` to your robotidy configuration you should be able to maintain your existing line feeds. Robotidy will of course normalize them (if you use different line feeds in the middle of the file it will be overwritten with what was autodiscovered in the first line of the file).

**Comment by bhirsz** [comment]
```
IF    "${s}"=="FAIL"    RETURN    ${False}
```
is perfectly valid syntax (it's inline IF syntax introduced in 5.0). The problem is that it seems you're running the transformed robot file with robot version < 5.0. Either you need to run robotidy on the same version you're using to run your robot files, or you can use ``robotidy --target-version rf4`` configuration to disable syntax that is not supported in version below RF 5.0.

If it doesn't help or there is any other issue let me know.

**Comment by yahman72** [comment]
> wrt. "lineseparator"

thanks, got it. 

I did not check the `robotidy --help` just searched in https://robotidy.readthedocs.io but the search did not return any hits for "lineseparator", "linefeed" --> small documentation bug?

regarding the default: I would consider changing the default to `auto` because majority of users will be using git to store the robot files. And git has different ways of handling them (e.g. https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings) --> defaulting to `native` could lead to unexpected results, like in my case, when working with git.
In short: `auto` would sound like a safer choice.



> wrt.
IF    "${s}"=="FAIL"    RETURN    ${False}

this is weird, I'm using RF 5.0.1
```
> pip list | egrep robotfram
robotframework                    5.0.1
```
but cannot reproduce the problem now - maybe I messed up something with the used virtualenvs and the venv was using an older RF ... 
--> we can ignore this part

## conclusion

This issue can be closed - unless you want to keep it open to fix that documentation bug and/or change lineseparator default value.

**Comment by bhirsz** [comment]
Thanks for the update. I will keep the thread open because I want to update our documentation. I will also think about default for lineseperator. I agree with your reasoning, and auto looks like a safer choice. It was added later while native was already on place for long time that's why it wasn't default option from the start.

**Comment by bhirsz** [comment]
The behaviour is described in https://robotidy.readthedocs.io/en/stable/quickstart.html#line-endings . I will close the issue for now (but I will take auto option as a default for consideration)
